src/hammer-vlsi/synthesis/vivado/__init__.py

```diff
@@ -1,33 +1,157 @@
-from hammer_vlsi import HammerSynthesisTool, HammerToolStep, deepdict
-
 from typing import Dict, List, Any
 
 import os
+import errno
+import pkg_resources
+from itertools import chain
+
+from jinja2 import Template
+
+from hammer_vlsi import HammerSynthesisTool, HammerToolStep, deepdict
+
+
+def _sym_link_force(source: str, dest: str) -> None:
+    try:
+        os.symlink(source, dest)
+    except OSError as e:
+        if e.errno == errno.EEXIST:
+            os.remove(dest)
+            os.symlink(source, dest)
+        else:
+            raise e
+
+
+class VivadoCommon(HammerSynthesisTool):
+
+    _temp_dir = 'file_templates'
+    _base_files = ['board.tcl', 'paths.tcl', 'project.tcl']
+    synth_tag = 'vivado_synth'
```

Comment:
I think we can probably inline some/all of these unless there's extensive references to it?

In particular `synth_tag` is redundant since Hammer will autopopulate `self.run_dir` to be (by default) either "obj/syn-rundir" or "obj/par-rundir".